### PR TITLE
feat: Flinksql new tutorial stream stream join

### PR DESCRIPTION
**Pull request by bbejeck**
### Description

[asana](https://app.asana.com/0/1201951209538300/1203941696282748/f)

### Staging Docs

[flinksql-interval-joins](http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/feat_flinksql_new_tutorial_stream_stream_join/join-a-stream-to-a-stream/flinksql.html)

### CODE REVIEW

**Comment by bbejeck** [comment]
test is failing and something is wrong with the layout - I'm looking into it

File: _includes/tutorials/joining-stream-stream/flinksql/markup/dev/transient-join.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Let's join these event streams together to gain some insight into the order-to-shipping process.  Then we'll discuss some of the concepts used in the query to enable retrieving your desired results.
+NOTE: There are several columns in the results so it's best to set the result mode to `TABLEAU` using the following command:
+++++
+<pre class="snippet"><code class="sql"> SET sql-client.execution.result-mode=tableau; </code></pre>
```
Comment (davetroiano) [inline _includes/tutorials/joining-stream-stream/flinksql/markup/dev/transient-join.adoc:4]: it's unfortunate that hljs doesn't do the right thing here and treats `sql` and `mode` as keywords and makes them bold:

image

Comment (davetroiano) [reply _includes/tutorials/joining-stream-stream/flinksql/markup/dev/transient-join.adoc:4]: maybe use `plaintext` here instead. .. though I'm wondering how widespread an issue this is. could be a separate task to sweep the rest of KT

File: _includes/tutorials/joining-stream-stream/flinksql/markup/dev/validate-shipped-orders-table.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+<pre class="snippet"><code class="sql">{% include_raw tutorials/joining-stream-stream/flinksql/code/tutorial-steps/dev/print-output-topic.sql %}</code></pre>
++++++
+
+This will yield the same output that the transient query did (perhaps in a different order)
+
++++++
+<pre class="snippet"><code class="shell">{% include_raw tutorials/joining-stream-stream/flinksql/code/tutorial-steps/dev/expected-shipped-orders.log %}</code></pre>
++++++
+
+We could also query the underlying topic directly using `console-consumer`. Open a new terminal window and run the following command:
```
Comment (davetroiano) [inline _includes/tutorials/joining-stream-stream/flinksql/markup/dev/validate-shipped-orders-table.adoc:20]: using `kafka-console-consumer`

File: _includes/tutorials/joining-stream-stream/flinksql/markup/test/create-resource-create-shipments.sql.template.adoc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+We could always inline the SQL statements in our Java test code, but creating separate resource files makes our test more readable and easier to maintain. Further, we can imagine parametrizing URLs as well so that we can have a single set of source controlled queries to use in tests as well as staging or production environments.
+
+There are a handful of resources to create for our test. These mirror the queries that we developed earlier.
+
+Create the following file at `src/test/resources/create-shipments.sql.template`. Note the `KAFKA_PORT` placeholder in this file. Out test will dynamically assign these to the ports that Testcontainers assigns.
```
Comment (davetroiano) [inline _includes/tutorials/joining-stream-stream/flinksql/markup/test/create-resource-create-shipments.sql.template.adoc:5]: Out test --> Our test

File: _includes/tutorials/joining-stream-stream/flinksql/markup/test/create-resource-shipped-orders.sql.adoc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Next, create the table for the join result with the following file at `src/test/resources/create-shipped-orders.sql.template`:
++++++
+<pre class="snippet"><code class="sql">{% include_raw tutorials/joining-stream-stream/flinksql/code/src/test/resources/create-shipped-orders.sql.template %}</code></pre>
++++++
+
+Note again we specify the bootstrap servers configuration port with `KAFKA_PORT` which our test code will replace with actual port.
```
Comment (davetroiano) [inline _includes/tutorials/joining-stream-stream/flinksql/markup/test/create-resource-shipped-orders.sql.adoc:6]: with the actual port

**Review by davetroiano — APPROVED** [review]
nice job on this @bbejeck ... few minor comments

**Comment by bbejeck** [comment]
Failures unrelated.

The test for this [PR passed](https://semaphore.ci.confluent.io/jobs/cf4a70eb-a48d-40ed-b19c-46c707b4e7de).

**Pull request merged**